src/wallet/wallet.cpp

```diff
@@ -585,6 +585,19 @@ void CWallet::ChainTipAdded(const CBlockIndex *pindex,
 {
     IncrementNoteWitnesses(pindex, pblock, sproutTree, saplingTree);
     UpdateSaplingNullifierNoteMapForBlock(pblock);
+
+    // SetBestChain() can be expensive for large wallets, so do this
+    // at most once per hour; the wallet state will be brought up to
+    // date during rescanning on startup.
```

Comment:
How long does this actually take for a large wallet?

Reply:
I don't have a good sense of what a large wallet is (despite my comment). I wrote a python test to fill a wallet with 1000 transactions with notes, and added timing measurement code to zcashd that shows `SetBestChain()` takes about 1900ms on my laptop. (I do a larger experiment and report back here later.)

That's probably low enough that we could do this more often than once per hour, but it's been that way for a long time.